cluster/cluster.go

```diff
@@ -563,3 +571,116 @@ func MustGetSegmentConfiguration(connection *dbconn.DBConn, getMirrors ...bool)
 	gplog.FatalOnError(err)
 	return segConfigs
 }
+
+/*GetSegmentConfigurationFromFile to parse gpsegconfig_dump or similar file to retrieve segment configuration information.
+gpsegconfig_dump file gets created at $COORDINATOR_DATA_DIR/gpsegconfig_dump by fts process
+The gpsegconfig_dump file follows a structured format, as illustrated in the example below:
+1 -1 p p n u 6000 localhost localhost /data/temp1
+2 0 p p n u 6002 localhost localhost /data/temp2
+3 1 p p n u 6003 localhost localhost /data/temp3
+4 2 p p n u 6004 localhost localhost /data/temp4
+
+Recommended use of the api is to get populate the gpdbArray when database is down.
+if the database is up use GetSegmentConfiguration()/MustGetSegmentConfiguration()  API
+
+input:
+confFile - absolute path of the gpsegconfig_dump or similar structured file
+(e.g. /Users/shrakesh/workspace/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/gpsegconfig_dump)
+
+return:
+Slice of populated SegConfig objects or else nil
+error if any or else nil.*/
+
+func GetSegmentConfigurationFromFile(confFile string) ([]SegConfig, error) {
+
+	/*Check if the given argument confFile is empty*/
+	if len(strings.TrimSpace(confFile)) == 0 {
+		errStr := fmt.Sprintf("Error: Given configuration file name is empty")
+		gplog.Error(errStr)
```

Comment:
Why are we logging all the errors with gplog ? Isn't that the utility's responsibility ?

Reply:
yes, that's a good point thanks,  i will remove the logging.